Userland/Libraries/LibWeb/Fetch/BodyInit.cpp

```diff
@@ -133,11 +141,29 @@ WebIDL::ExceptionOr<Infrastructure::BodyWithType> extract_body(JS::Realm& realm,
 
     // FIXME: 11. If source is a byte sequence, then set action to a step that returns source and length to source’s length.
     // For now, do it synchronously.
-    if (source.has<ByteBuffer>())
-        length = source.get<ByteBuffer>().size();
+    if (source.has<ByteBuffer>()) {
+        action = JS::create_heap_function(realm.heap(), [source]() {
+            auto const* byte_buffer = source.get_pointer<ByteBuffer>();
+            return ActionResult { .source = *byte_buffer, .length = byte_buffer->size() };
+        });
```

Comment:
FYI it took me a hot minute to wrap my head around the awkward spec text wording here, but I think this is meant to be 2 individual steps:
> a. Set action to a step that returns source
> b. *And* set length to source's length

Because `length` is needed synchronously, when we create the body below. My local copy of these steps look like this:

```c++
// 6. Let action be null.
Function<ByteBuffer()> action;

// 11. If source is a byte sequence, then set action to a step that returns source and length to source’s length.
if (source.has<ByteBuffer>()) {
    action = [source = MUST(ByteBuffer::copy(source.get<ByteBuffer>()))]() mutable {
        return move(source);
    };

    length = source.get<ByteBuffer>().size();
}

// 12. If action is non-null, then run these steps in parallel:
if (action) {
    Platform::EventLoopPlugin::the().deferred_invoke([&realm, stream, action = move(action)] {
        HTML::TemporaryExecutionContext execution_context { Bindings::host_defined_environment_settings_object(realm), HTML::TemporaryExecutionContext::CallbacksEnabled::Yes };

        // 1. Run action.
        auto bytes = action();

        // Whenever one or more bytes are available and stream is not errored, enqueue the result of creating a
        // Uint8Array from the available bytes into stream.
        if (!bytes.is_empty() && !stream->is_errored()) {
            auto array_buffer = JS::ArrayBuffer::create(stream->realm(), move(bytes));
            auto chunk = JS::Uint8Array::create(stream->realm(), array_buffer->byte_length(), *array_buffer);

            Streams::readable_stream_enqueue(*stream->controller(), chunk).release_value_but_fixme_should_propagate_errors();
        }

        // When running action is done, close stream.
        // Streams::readable_stream_close(*stream);
    });
}
```

Reply:
Amended the commit according to your suggestions here.